core/src/wallet/bitcoin/BitcoinLikeAccount.cpp

```diff
@@ -260,6 +260,16 @@ namespace ledger {
             }
         }
 
+        std::vector<std::shared_ptr<api::Address>> BitcoinLikeAccount::fromBitcoinAddressesToAddresses(const std::vector<std::shared_ptr<BitcoinLikeAddress>> &addresses) {
+            AbstractAccount::AddressList result(addresses.size());
+            auto i = 0;
+            for (auto& addr : addresses) {
+                result[i] = std::dynamic_pointer_cast<api::Address>(addr);
+                i += 1;
+            }
```

Comment:
I think you can simplify that function by using `reserve` and `push_back`:

```suggestion
AbstractAccount::AddressList result;
result.reserve(addresses.size());

for (auto const& addr : addresses) {
  result.push_back(std::dynamic_pointer_cast<api::Address>(addr);
}
```

That way, you don’t have to keep the logic of this `i` variable, and you still make only one allocation.